test/astwalker.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+var tape = require('tape')
+var AstWalker = require('../src/util/astWalker')
+var node = require('./resources/ast')
+
+tape('ASTWalker', function (t) {
+  t.test('ASTWalker.walk', function (st) {
+    st.plan(24)
+    var astwalker = new AstWalker()
+
+    astwalker.walk(node.ast.AST, function (error, node) {
+      if (error) {
+        st.fail(error)
+      } else {
+        if (node.name === 'ContractDefinition') {
+          checkContract(st, node)
+        }
+        if (node.name === 'FunctionDefinition') {
+          checkSetFunction(st, node)
+        }
+      }
+      return true
+    })
+
+    var callback = {}
+    callback.FunctionDefinition = function (error, node) {
+      if (error) {
+        st.fail(error)
+      } else {
+        st.equal(node.name, 'FunctionDefinition')
+        st.equal(node.attributes.name === 'set' || node.attributes.name === 'get', true)
+      }
+    }
+    astwalker.walk(node.ast.AST, callback)
+  })
+})
+
+function checkContract (st, node) {
+  st.equal(node.attributes.name, 'test')
+  st.equal(node.children[0].attributes.name, 'x')
```

Comment:
this function is called in the callback of the walk function so it's also testing the walker